src/aws.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @typedef {object} ParsedTopic
+ * @property {string|undefined} region - aws region of topic
+ */
+
+/**
+ * Parse an AWS SNS TopicArn
+ * @param {string} topicArn - e.g. arn:aws:sns:us-east-2:foo:topicName
+ * @returns {ParsedTopic}
+ */
+function parseTopic(topicArn) {
```

Comment:
Why not use https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_util_arn_parser.html here?

Reply:
@alanshaw this was the fastest way to make the test pass. also all the aws modules just feel like a lot to reach for sometimes vs a regex or something simple at first. Totally happy to use that fn though. Thanks for suggesting it. I'll pull it in.

Reply:
started using that [@aws-sdk/util-arn-parser](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_util_arn_parser.html)
https://github.com/elastic-ipfs/bucket-to-indexer/pull/13/commits/1832907b9bcb3f0f07c798a0b3166436129550c3